flaky test SALChannelTest.testBasicRead

```
16:49:07 Test Case 'SALChannelTest.testBasicRead' started at 2020-01-28 16:49:07.039
16:49:12 /code/Tests/NIOTests/SyscallAbstractionLayer.swift:511: error: SALChannelTest.testBasicRead : XCTAssertNoThrow failed: threw error "TimeoutError(description: "u2k")" - 
16:49:12 Test Case 'SALChannelTest.testBasicRead' failed (5.008 seconds)
```

--- a/Tests/NIOTests/SyscallAbstractionLayer.swift
+++ b/Tests/NIOTests/SyscallAbstractionLayer.swift
@@ -508,7 +508,10 @@ extension SALTest {
             XCTAssertNil(error, "unexpected error: \(error!)")
             group.leave()
         })
-        XCTAssertNoThrow(try self.assertWakeup())
+        // We're in a slightly tricky situation here. We don't know if the EventLoop thread enters `whenReady` again
+        // or not. If it has, we have to wake it up, so let's just put a return value in the 'kernel to user' box, just
+        // in case :)
+        XCTAssertNoThrow(try self.kernelToUserBox.waitForEmptyAndSet(.returnSelectorEvent(nil)))
         group.wait()
 
         self.group = nil

